src/systems/a320_systems/src/lib.rs

```diff
@@ -122,7 +122,7 @@ impl Aircraft for A320 {
             &self.apu_overhead,
             &self.engine_fire_overhead,
             [&self.engine_1, &self.engine_2],
-            &self.hydraulic,
+            self.hydraulic.generator_control_unit(),
```

Comment:
**suggestion**: This violates the Law of Demeter. Maybe `A320Hydraulic` can expose the necessary trait and handle passing it to the GCU internally?